src/stream/src/executor/aggregation/approx_count_distinct.rs

```diff
@@ -0,0 +1,366 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module implements `StreamingApproxCountDistinct`.
+
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+
+use itertools::Itertools;
+use risingwave_common::array::stream_chunk::Ops;
+use risingwave_common::array::*;
+use risingwave_common::buffer::Bitmap;
+use risingwave_common::error::{ErrorCode, Result};
+use risingwave_common::types::{Datum, DatumRef, Scalar, ScalarImpl};
+
+use super::StreamingAggStateImpl;
+
+const INDEX_BITS: u8 = 14; // number of bits used for finding the index of each 64-bit hash
+const NUM_OF_REGISTERS: usize = 1 << INDEX_BITS; // number of registers available
+const COUNT_BITS: u8 = 64 - INDEX_BITS; // number of non-index bits in each 64-bit hash
+
+// Approximation for bias correction for 16384 registers. See "HyperLogLog: the analysis of a
+// near-optimal cardinality estimation algorithm" by Philippe Flajolet et al.
+const BIAS_CORRECTION: f64 = 0.72125;
+
+#[derive(Clone, Debug)]
+struct RegisterBucket {
+    count_1_to_16: Vec<u32>,
+    count_17_to_24: Vec<u16>,
+    count_25_to_32: Vec<u8>,
+    count_33_to_64: u32,
+}
+
+impl RegisterBucket {
```

Comment:
Just some nits:
1. If each element is going to be a register, maybe this should be `RegisterBuckets`?
2. It seems easier to understand that each location you hash into is a "bucket", whereas each count is a "register". Not sure if you agree! It does make it inconsistent terminology with the batch case.

Reply:
I think it's better to keep the terminology consistent with the batch case and the other HyperLogLog implementations